bundles/automation/org.eclipse.smarthome.automation.event.test/src/test/groovy/org/eclipse/smarthome/automation/event/RuleEventTest.groovy

```diff
@@ -0,0 +1,170 @@
+ */
+package org.eclipse.smarthome.automation.event;
+
+
+import static org.junit.Assert.*
+
+import static org.hamcrest.CoreMatchers.*
+import static org.junit.Assert.*
+import static org.junit.matchers.JUnitMatchers.*
+
+import org.eclipse.smarthome.automation.Action
+import org.eclipse.smarthome.automation.Condition
+import org.eclipse.smarthome.automation.Rule
+import org.eclipse.smarthome.automation.RuleRegistry
+import org.eclipse.smarthome.automation.Trigger
+import org.eclipse.smarthome.automation.events.RuleAddedEvent
+import org.eclipse.smarthome.automation.events.RuleRemovedEvent
+import org.eclipse.smarthome.automation.events.RuleStatusInfoEvent
+import org.eclipse.smarthome.automation.events.RuleUpdatedEvent
+import org.eclipse.smarthome.core.autoupdate.AutoUpdateBindingConfigProvider
+import org.eclipse.smarthome.core.events.Event
+import org.eclipse.smarthome.core.events.EventPublisher
+import org.eclipse.smarthome.core.events.EventSubscriber
+import org.eclipse.smarthome.core.items.ItemProvider
+import org.eclipse.smarthome.core.items.ItemRegistry
+import org.eclipse.smarthome.core.items.events.ItemEventFactory
+import org.eclipse.smarthome.core.items.events.ItemStateEvent
+import org.eclipse.smarthome.core.items.events.ItemUpdatedEvent
+import org.eclipse.smarthome.core.library.items.SwitchItem
+import org.eclipse.smarthome.core.library.types.OnOffType
+import org.eclipse.smarthome.core.types.Command
+import org.eclipse.smarthome.core.types.TypeParser
+import org.eclipse.smarthome.test.OSGiTest
+import org.eclipse.smarthome.test.storage.VolatileStorageService
+import org.junit.Before
+import org.junit.Test
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import com.google.common.collect.Sets
+
+/**
+ * This tests events of rules
+ * 
+ * @author Benedikt Niehues - initial contribution
+ *
+ */
+class RuleEventTest extends OSGiTest{
+
+    final Logger logger = LoggerFactory.getLogger(RuleEventTest.class)
+    VolatileStorageService volatileStorageService = new VolatileStorageService()
+
+    @Before
+    void before() {
+        def itemProvider = [
+            getAll: {
+                [
+                    new SwitchItem("myMotionItem"),
+                    new SwitchItem("myPresenceItem"),
+                    new SwitchItem("myLampItem"),
+                    new SwitchItem("myMotionItem2"),
+                    new SwitchItem("myPresenceItem2"),
+                    new SwitchItem("myLampItem2")
+                ]
+            },
+            addProviderChangeListener: {},
+            removeProviderChangeListener: {},
+            allItemsChanged: {}] as ItemProvider
+        registerService(itemProvider)
+        registerService(volatileStorageService)
+
+        def autoupdateConfig = [
+            autoUpdate: { String itemName -> return true }
+
+        ] as AutoUpdateBindingConfigProvider
+        registerService(autoupdateConfig)
+    }
+
+    @Test
+    public void testRuleEvents() {
```

Comment:
I don´t get what the test is doing? I think we should check, if a rule is added the "RuleAddedEvent" is posted. Then if a rule is modified, the modified event is sent, etc. Same for rule removed.

For RuleStatusInfo events, the test has to be a bit more complex I guess. We should check a rule is uninitialized as long as there is no ModuleType. And then we have to check that is smoothly changes the status.